lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -4108,34 +4108,60 @@ bool SILParser::parseSpecificSILInstruction(SILBuilder &B,
 
     break;
   }
-  case SILInstructionKind::AllocStackInst:
-  case SILInstructionKind::MetatypeInst: {
-
+  case SILInstructionKind::AllocStackInst: {
     bool hasDynamicLifetime = false;
-    if (Opcode == SILInstructionKind::AllocStackInst &&
-        parseSILOptional(hasDynamicLifetime, *this, "dynamic_lifetime"))
+    bool isLexical = false;
+
+    while (P.consumeIf(tok::l_square)) {
+      Identifier ident;
+      SourceLoc identLoc;
+      if (parseSILIdentifier(ident, identLoc,
+                             diag::expected_in_attribute_list)) {
+        if (P.consumeIf(tok::r_square)) {
+          continue;
+        } else {
+          return true;
```

Comment:
@nate-chandler 
PSA: Never just return true in the parser without issuing a diagnostic! We won't even get line numbers when parsing fails.

For all of this, you should just be able to use
`parseSILOptional(StringRef &Result, SourceLoc &Loc, SILParser &SP`

Reply:
https://github.com/apple/swift/pull/39350